app.js

```
@@ -0,0 +1,100 @@
+//             res.send({error: "Please fill out all the information or Class already exists",
+//             timestamp: "YYYY, MM/DD HH:MM:SS"})
+//         }else {
+//             mySchool.classes[mySchool.addClass(req.params.name, req.params.teacher)]
+//             mySchool.classes[req.params.name]["message"] = "Created a new class"
+//             mySchool.classes[req.params.name]["timestamp"] = "Todays Date"
+//             res.send(mySchool.classes) 
+//         }
+// })
+app.post("/class/:className/enroll", (req, res) => {
+    // let className = req.params.className
+    let values = Object.values(req.body)
+    console.log("values", values)
+    
+    let newStudent = new Student(req.body.name, req.body.age, req.body.city, req.body.grade)
+    
+    if(values.includes("")) {
+        console.log("Please fill out all the information for the student");
+        
+        res.json({
+            error: "Please fill out all the information for the student",
+            timestamp: "YYYY, MM/DD HH:MM:SS"
+        })
+    } else {
+        mySchool.enrollStudent(req.params.className, newStudent)
+        console.log("classes", mySchool.classes);
+        console.log("new student", newStudent)
+        
+        res.json({
+            studentEnrolled: newStudent,
+            message: "Enrolled Student",
+            timestamp: "YYYY, MM/DD HH:MM:SS"
+        })
+    }
+})
+
+app.get("/class/:className/students", (req,res) => {
+    console.log("my body",req.body);
+    console.log("my params", req.params.className);
+    mySchool.classes[req.params.className].students
+    let studentsArr = mySchool.getStudentsByClassWithFilter(req.params.className, req.body.failing, req.body.city)
+    
+    debugger
+    res.json({
+        students: studentsArr,
+        message: "thank you"
+    })
+    mySchool.classes[mySchool.addClass("math", "brandon")]
+    let className = req.params.className
+    res.send(mySchool.classes[className]["students"])
```

Having this here will give you a silent error since you're trying to give a response after already sending a response beforehand.